src/Common/PODArray.h

```diff
@@ -104,8 +105,14 @@ class PODArrayBase : private boost::noncopyable, private TAllocator    /// empty
     char * c_end_of_storage = null;    /// Does not include pad_right.
 
     /// The amount of memory occupied by the num_elements of the elements.
-    static size_t byte_size(size_t num_elements) { return num_elements * ELEMENT_SIZE; }
-
+    static size_t byte_size(size_t num_elements) 
+    {
+        if (num_elements > SIZE_MAX/ELEMENT_SIZE) 
```

Comment:
Better to use `__builtin_mul_overflow` for this purpose.

Reply:
Hi Alexey, I have updated the code as per your suggestion.  Thanks